### DISCUSSION TITLE
Cryptic error, help with how to track down issue

### DISCUSSION THREAD

**Discussion by rajivkaggarwal** [vercel/community#2022] [Help] [answered]
### Summary

I am new to Vercel.  I created a Nuxt3 project in Vercel from one of their templates.

I then made some edits to introduce Supabase and Tailwind.  These changes work fine locally.  I then committed the changes to GitHub and Vercel automatically built and deployed.

I see no errors in the Vercel build logs.

Upon accessing via web browner I see this in the Vercel logs:

```
TypeError: Cannot read properties of undefined (reading 'bind')
    at (index.mjs:1:392139)
    at (__nitro:middleware.js:1:17)
```

![Screen Shot 2023-04-09 at 1 50 52 PM](https://user-images.githubusercontent.com/1007527/230788588-629095a6-2777-46f0-9c96-a5b2ecf42f0f.png)

![Screen Shot 2023-04-09 at 1 51 28 PM](https://user-images.githubusercontent.com/1007527/230788596-ccfca35a-c7bf-4041-ba2e-0b75efdf0719.png)

**Comment by rajivkaggarwal** [comment, marked as answer]
My nuxt.config.ts file, which was auto-generated by the template on Vercel, contained the following:

```
  nitro: {
    preset: 'vercel-edge',
  },
```

I changed 'vercel-edge' to 'vercel' and now it works...

There doesn't seem to be any mention of this "up front" when choosing a template and generating a new project.  Would be helpful if there was some disclaimer/limitation mentioned...

In addition, the [comparison of serverless to edge](https://vercel.com/docs/concepts/limits/overview#functions-comparison) doesn't mention anything explicit other than this regarding Runtime and Node.js APIs:

<img width="627" alt="Screen Shot 2023-04-09 at 2 15 54 PM" src="https://user-images.githubusercontent.com/1007527/230789716-fb298475-b663-4689-95e1-1fafc9ceaf51.png">


Is limited support for Node.js APIs the reason I can't use Edge?

**Reply by amyegan** [reply]
I believe your hypothesis is correct. Edge functions use Vercel's [Edge Runtime](https://edge-runtime.vercel.sh/packages/runtime) (built on [V8 JavaScript and WebAssembly engine](https://developers.google.com/apps-script/guides/v8-runtime)), which has limited support for Node.js APIs. See [Edge Functions Limitations](https://vercel.com/docs/concepts/functions/edge-functions/limitations#) for more info.

If you let me know which template you used, I will create a ticket to include more helpful warnings about these limitations.
